apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,80 @@
+import { IPreferenceChannels, StepTypeEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      workflowId: template.name,
+      tags: template.tags,
+      active: template.active,
+      payload: template.data ? { schema: template.data } : undefined, // Assuming data is mapped to payload schema
+      controls: template.payloadSchema ? { schema: template.payloadSchema } : undefined,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      code: template.name, // Assuming name is used as code
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      workflowId: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepSummery: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
```

Comment:
I think the name is not informative. 
```suggestion
      stepsType: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
```

What is the use case for having this list of types in the DTO?

Reply:
stepTypeOverviews